sherpa/plot/backends.py

```diff
@@ -0,0 +1,995 @@
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+        '''
+        pass
+
+    def end(self):
+        '''Called from the UI after an interactive plot is done.
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+        '''
+        pass
+
+    @add_kwargs_to_doc(kwargs_doc)
+    @translate_args
+    def plot(self, x, y, *,
+             xerr=None, yerr=None,
+             title=None, xlabel=None, ylabel=None,
+             xlog=False, ylog=False,
+             overplot=False, clearwindow=True,
+             label=None,
+             xerrorbars=False,
+             yerrorbars=False,
+             color=None,
+             linestyle='solid',
+             linewidth=None,
+             marker='None',
+             alpha=None,
+             markerfacecolor=None,
+             markersize=None,
+             **kwargs):
+        """Draw x, y data.
+
+        This method combines a number of different ways to draw x/y data:
+            - a line connecting the points
+            - scatter plot of symbols
+            - errorbars
+
+        All three of them can be used together (symbols with errorbars connected
+        by a line), but it is also possible to use only one or two of them. By
+        default, a line is shown (``linestyle='solid'``), but marker and error
+        bars are not (``marker='None'`` and ``xerrorbars=False`` as well as
+        ``yerrorbars=False``).
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+
+        Parameters
+        ----------
+        x : array-like or scalar number
+            x values
+        y : array-like or scalar number
+            y values, same dimension as `x`.
+        {kwargs}
+        """
+        pass
+
+    @add_kwargs_to_doc(kwargs_doc)
+    @translate_args
+    def histo(self, xlo, xhi, y, *,
+              yerr=None,
+              title=None, xlabel=None, ylabel=None,
+              overplot=False, clearwindow=True,
+              xlog=False, ylog=False,
+              label=None,
+              xerrorbars=False,
+              yerrorbars=False,
+              color=None,
+              linestyle='solid',
+              linewidth=None,
+              marker='None',
+              alpha=None,
+              markerfacecolor=None,
+              markersize=None,
+              ecolor=None,
+              capsize=None,
+              # barsabove=False,
+              **kwargs):
+        """Draw histogram data.
+
+        The histogram is drawn as horizontal lines connecting the
+        start and end points of each bin, with vertical lines connecting
+        consecutive bins. Non-consecutive bins are drawn with a
+        (NaN, NaN) between them so no line is drawn connecting them.
+
```

Comment:
Technically this discussion - of how non-cnosecutive bins are handled - is only valid for the matplotlib backend. Other backends might deal with this differently.

Reply:
Possibly - but again that seems to be common behaviour for those backends that I looked at.